### ISSUE TITLE
Error with return statement

### ISSUE THREAD

**Issue by KimBruce** [gracelang/minigrace#167]
Consider the following code:
```
var l: List<Number> := emptyList
l.add(1)
l.add(3)

method isLegal (tempPosition: List<Number>) -> Boolean {
        for (tempPosition) do {elt: Number ->
            if ((elt % 2) == 1) then {
                return false
            }
        }
        true
}

if (isLegal(l)) then {
    print "fine"
} else {
    print "not fine"
}
```
This code, which should be legal, results in the error message:
```
Internal error around line 14 of testReturn: [object Object]
[object Object]
```
It should instead print out "not fine"

I believe the error is in the use of the "return" statement in the for loop.

**Comment by kjx** [comment]
On 31/05/2016, at 7:08AM, Kim Bruce <notifications@github.com> wrote:
> 
> It should instead print out "not fine"

it does on kernan. 

unfortunately I had to hack the collections library to run on the official syntax on Kernan.. 
that file is lurking on the combinators repo...

James

**Comment by apblack** [comment]
The problem is in the type annotation on the argument to isLegal.  Remove it, and it compiles fine. 

I’ll try and see why ...

**Comment by apblack** [comment]
This is fixed in 984310b.  The problem was an interaction between the type test and the invocationCount global variable that keeps track of where a `return` should return to.
